packages/realm-react/src/RealmProvider.tsx

```diff
@@ -161,6 +191,45 @@ export function createRealmProvider(
   };
 }
 
+/**
+ * Generates a flexible `RealmProvider` which is either based on configuration or based on a realm
+ * depending on its props.
+ * @param RealmContext - The context that will contain the Realm instance
+ * @returns a RealmProvider component that provides context to all context hooks
+ */
+export function createFlexibleRealmProvider(RealmContext: React.Context<Realm | null>): FlexibleRealmProviderFC {
+  return ({ realm, ...restProps }) => {
+    return useMemo(() => {
```

Comment:
I think `useMemo` is appropriate here but may be worth discussing if there's better alternatives

Reply:
Hm, good point, I was under impression that tests were handling this case but now that I think about it none of them seem to be doing exactly this change